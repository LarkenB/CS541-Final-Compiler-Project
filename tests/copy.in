float x;
int y;
char z;
x = y = z = 4;