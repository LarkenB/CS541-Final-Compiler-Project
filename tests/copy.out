target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

define dso_local i32 @main() {
  %t1 = alloca float
  %t2 = alloca i32
  %t3 = alloca i8
  %t4 = trunc i32 4 to i8
  store i8 %t4, i8* %t3
  %t5 = sext i8 %t4 to i32
  store i32 %t5, i32* %t2
  %t6 = sitofp i32 %t5 to float
  store float %t6, float* %t1
  ret i32 0
}
